src/translate.directive.ts

```diff
@@ -53,6 +53,16 @@ export class TranslateDirective implements AfterViewChecked, OnDestroy {
         this.checkNodes();
     }
 
+    ngOnChanges(changes: any) { //Everytime an Input property is changed, Angular fires ngOnchanges
+        for (let propName in changes) {
+            if(propName === "translateParams") {//We only care if the translateParams changed
```

Comment:
If we only care if the translateParams changed, why do a loop through all of the properties in changes? I would instead remove the for loop and just have:

```
if (changes.translateParams) {
    'contents of original if block'
}
```